src/repositories/ads.repository.ts

```diff
@@ -0,0 +1,18 @@
+import { Repository } from '../core/repository';
+
+export class AdsRepository extends Repository {
+  public async getPostInsights(id: string) {
+    const { body } = await this.client.request.send({
+      url: '/api/v1/ads/graphql/?locale=en_US&vc_policy=insights_policy&surface=post',
```

Comment:
Query string shoud be placed in `qs: {}` parameter.
Why you have `locale` hardcoded to `en_US`?